### PR TITLE
add graph algos into guide

### PR DESCRIPTION
**Pull request by mneedham**

### CODE REVIEW

File: recommendations/index.html [hunk 1]
```diff
@@ -88,11 +88,10 @@ <h4>Graph-Based Recommendations</h4>
 </li>
 </ol>
 </div>
-<img src="http://guides.neo4j.com/sandbox/recommendations/img/title1.png" class="img-responsive">
+<img src="//localhost:8001/img/title1.png" class="img-responsive">
```
Comment (jexp) [inline recommendations/index.html]: probably not the right URL?

File: recommendations/index.html [hunk 2]
```diff
@@ -88,11 +88,10 @@ <h4>Graph-Based Recommendations</h4>
...
 </div>
 <div class="col-lg-4 ng-scope">
-<img src="images/click-next.png" style="position:relative;right:-23px;top:-56px;" class="pull-right">
-<img src="http://guides.neo4j.com/sandbox/recommendations/img/openmoviegraph.png" class="img-responsive">
+<img src="//localhost:8001/img/openmoviegraph.png" class="img-responsive">
```
Comment (jexp) [inline recommendations/index.html]: URL

File: recommendations/index.html [hunk 3]
```diff
@@ -227,7 +226,7 @@ <h4>Graph Patterns</h4>
 <p>Relationships are defined within square brackets <code>[]</code>. Optionally we can specify type and direction:</p>
 </div>
 <div class="paragraph">
-<p><code>(:Movie)<strong>&lt;-[:RATED]-</strong>(:User)</code></p>
```
Comment (jexp) [inline recommendations/index.html:230]: what was the issue here? just curious?

File: recommendations/index.html [hunk 4]
```diff
@@ -163,7 +162,7 @@ <h4>Eliminate Data Silos</h4>
 </div>
 </div>
 <div class="col-lg-8 ng-scope">
-<img src="http://guides.neo4j.com/sandbox/recommendations/img/datamodel.png" class="img-responsive">
+<img src="//localhost:8001/img/datamodel.png" class="img-responsive">
```
Comment (jexp) [inline recommendations/index.html]: URL

File: recommendations/intro.adoc [hunk 5]
```diff
@@ -652,8 +652,24 @@ WITH COUNT(m) AS numbermovies, SUM(x.rating * y.rating) AS xyDotProduct,
...
+----
+
+We can also compute this measure using the https://neo4j.com/docs/graph-algorithms/current/algorithms/similarity-cosine[Cosine Similarity algorithm^] in the Neo4j Graph Algorithms Library.
+
+.Find the users with the most similar preferences to Cynthia Freeman, according to cosine similarity function
+[source, cypher]
+----
+MATCH (p1:User {name: 'Cynthia Freeman'})-[x:RATED]->(movie)
+MATCH (p2:User)-[x2:RATED]->(movie) WHERE p2 <> p1
+WITH p1, p2, collect(x.rating) AS p1Ratings, collect(x2.rating) AS p2Ratings
```
Comment (jexp) [inline recommendations/intro.adoc:667]: should we collect the x.rating earlier to not have them explode according by the 2nd match?

File: recommendations/intro.adoc [hunk 6]
```diff
@@ -702,17 +718,35 @@ WITH u1, u1_mean, u2, avg(r.rating) AS u2_mean, ratings
...
 ----
 
+We can also compute this measure using the https://neo4j.com/docs/graph-algorithms/current/algorithms/similarity-pearson[Pearson Similarity algorithm^] in the Neo4j Graph Algorithms Library.
+
+.Find users most similar to Cynthia Freeman, according to the Pearson similarity function
+[source, cypher]
+----
+MATCH (p1:User {name: 'Cynthia Freeman'})-[x:RATED]->(movie:Movie)
+WITH p1, algo.similarity.asVector(movie, x.rating) AS p1Vector
+MATCH (p2:User)-[x:RATED]->(movie:Movie) WHERE p2 <> p1
```
Comment (jexp) [inline recommendations/intro.adoc]: should we use x2 here for clarity?  also in the later places and the 2nd list-comprehension?

File: recommendations/intro.adoc [hunk 7]
```diff
@@ -765,6 +799,26 @@ RETURN m.title, SUM( pearson * r.rating) AS score
...
 
+
+.kNN movie recommendation using Pearson similarity function
+[source, cypher]
+----
+MATCH (u1:User {name: 'Cynthia Freeman'})-[x:RATED]->(movie:Movie)
+WITH u1, algo.similarity.asVector(movie, x.rating) AS u1Vector
+MATCH (u2:User)-[x:RATED]->(movie:Movie) WHERE u2 <> u1
+
+WITH u1, u2, u1Vector, algo.similarity.asVector(movie, x.rating) AS u2Vector
```
Comment (jexp) [inline recommendations/intro.adoc]: same here about x vs. x2

File: recommendations/intro.adoc (hunk 5 above)
Comment (mneedham) [reply recommendations/intro.adoc:667]: we can't - they need to refer to the same `movie` node. I'll move it onto one line so it's more clear what the query does.

**Pull request merged**